google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/jpa/test/BaseEntityManagerFunctionalTestCase.java

```diff
@@ -286,16 +270,43 @@ private void releaseUnclosedEntityManager(EntityManager em) {
 
 		if ( em.getTransaction().isActive() ) {
 			em.getTransaction().rollback();
-            log.warn("You left an open transaction! Fix your test case. For now, we are closing it for you.");
+			System.out.println(
+					"You left an open transaction! Fix your test case. For now, we are closing it for you.");
 		}
 		if ( em.isOpen() ) {
 			// as we open an EM before the test runs, it will still be open if the test uses a custom EM.
 			// or, the person may have forgotten to close. So, do not raise a "fail", but log the fact.
 			em.close();
-            log.warn("The EntityManager is not closed. Closing it.");
+			System.out.println("The EntityManager is not closed. Closing it.");
+		}
+	}
+
+	protected void doIfNotInitialized(Runnable runnable) {
+		if (!isInitialized) {
+			runnable.run();
+			isInitialized = true;
 		}
 	}
 
+	@Before
```

Comment:
Does `@BeforeClass` not work here? Not too big of a fan of static variable null-check.

Reply:
Hmm I tried making it BeforeClass but there are quite a lot of functions that are called along the line that all need to be made static as a result :( I think this would be preferable to that... What do you think? It ends up being like a dozen functions and members that need to be made to be static

Reply:
Ah I see, that's fine then, you can just do it your current way.